### PR TITLE
Various fixes

### LINKED ISSUE #5057: IPv8 import conflict with AnyDex
When running the tunnel helper script on our infrastructure (latest `devel` commit), I get the following error:

```
Jan 17 17:17:37 prxmx3 python3[28437]: Traceback (most recent call last):
Jan 17 17:17:37 prxmx3 python3[28437]:   File "src/tribler-core/run_tribler_headless.py", line 16, in <module>
Jan 17 17:17:37 prxmx3 python3[28437]:     from tribler_core.session import Session
Jan 17 17:17:37 prxmx3 python3[28437]:   File "/opt/tribler/src/tribler-core/tribler_core/session.py", line 47, in <module>
Jan 17 17:17:37 prxmx3 python3[28437]:     from tribler_core.modules.metadata_store.gigachannel_manager import GigaChannelManager
Jan 17 17:17:37 prxmx3 python3[28437]:   File "/opt/tribler/src/tribler-core/tribler_core/modules/metadata_store/gigachannel_manager.py", line 4, in <module>
Jan 17 17:17:37 prxmx3 python3[28437]:     from ipv8.taskmanager import TaskManager, task
Jan 17 17:17:37 prxmx3 python3[28437]: ImportError: cannot import name 'task' from 'ipv8.taskmanager' (/opt/tribler/src/anydex/anydex/../pyipv8/ipv8/taskmanager.py)
```

Most likely, the IPv8 of AnyDex gets prioritised over the one that Tribler provides.

### LINKED ISSUE #5058: Error when trying anonymous downloading (on testnet)
I just updated our tunnel infrastructure and I observe the following error when trying to perform anonymous downloading:

```
Jan 17 19:51:38 prxmx3 python3[5915]: Traceback (most recent call last):
Jan 17 19:51:38 prxmx3 python3[5915]:   File "/opt/tribler/src/pyipv8/ipv8/community.py", line 324, in on_packet
Jan 17 19:51:38 prxmx3 python3[5915]:     result = handler(source_address, data)
Jan 17 19:51:38 prxmx3 python3[5915]:   File "/opt/tribler/src/pyipv8/ipv8/lazy_community.py", line 39, in wrapper
Jan 17 19:51:38 prxmx3 python3[5915]:     return func(self, Peer(auth.public_key_bin, source_address), *unpacked)
Jan 17 19:51:38 prxmx3 python3[5915]:   File "/opt/tribler/src/pyipv8/ipv8/messaging/anonymization/community.py", line 927, in on_destroy
Jan 17 19:51:38 prxmx3 python3[5915]:     self.remove_exit_socket(circuit_id, "Got destroy")
Jan 17 19:51:38 prxmx3 python3[5915]:   File "/opt/tribler/src/tribler-core/tribler_core/modules/tunnel/community/triblertunnel_community.py", line 428, in remove_exit_socket
Jan 17 19:51:38 prxmx3 python3[5915]:     self.tribler_session.notifier.notify(NTFY.TUNNEL_REMOVE, exit_socket, additional_info)
Jan 17 19:51:38 prxmx3 python3[5915]:   File "/opt/tribler/src/tribler-core/tribler_core/notifier.py", line 28, in notify
Jan 17 19:51:38 prxmx3 python3[5915]:     callback(*args)
Jan 17 19:51:38 prxmx3 python3[5915]: TypeError: circuit_removed() missing 2 required positional arguments: 'circuit' and 'additional_info'
```

The only nodes deployed are on testnet and controlled by me. They run the latest commit on `devel`.

This seems to be some kind of error in the notifier?

### PR DESCRIPTION
**Pull request by devos50**
Fixes #5057, fixes #5058

### CODE REVIEW

**Comment by ichorid** [comment]
Do you really *have* to include the IPv8 submodule in AnyDex?

**Comment by devos50** [comment]
@ichorid technically, I can remove it from Anydex and rely on a system-wide `IPv8` version. It requires some changes to the Anydex testers/builders though. If this is the preferred solution, I will make the required changes.

**Comment by ichorid** [comment]
> @ichorid technically, I can remove it from Anydex and rely on a system-wide `IPv8` version. It requires some changes to the Anydex testers/builders though. If this is the preferred solution, I will make the required changes.

Yes, that will be another safety gauge. It is always safer to have flat dependencies list and explicit import procedures.  And of course, keep https://github.com/Tribler/tribler/pull/5059/commits/36df6013693be3bfd1a38c3dd7c7143572ac97cd in this PR.

**Comment by devos50** [comment]
Consider it done 👍

File: src/run_tribler.py [hunk 1]
```diff
@@ -66,8 +66,9 @@ def shutdown(session, *_):
 
         config.set_http_api_port(int(api_port))
         # If the API key is set to an empty string, it will remain disabled
-        if config.get_http_api_key() != '':
+        if config.get_http_api_key() != '' and config.get_http_api_key() != api_key:
```
Comment (ichorid) [inline src/run_tribler.py]: why not do `if config.get_http_api_key() not in ('', api_key):` ?

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Pull request merged**